Mathlib/Tactic/DefEqTransformations.lean

```diff
@@ -17,6 +17,31 @@ namespace Mathlib.Tactic
 
 open Lean Meta Elab Elab.Tactic
 
+/--
+This is `Lean.MVarId.changeLocalDecl` but makes sure to preserve local variable order.
+-/
+def _root_.Lean.MVarId.changeLocalDecl' (mvarId : MVarId) (fvarId : FVarId) (typeNew : Expr)
+    (checkDefEq := true) : MetaM MVarId := do
+  mvarId.checkNotAssigned `changeLocalDecl
+  let lctx := (← mvarId.getDecl).lctx
+  let some decl := lctx.find? fvarId | throwTacticEx `changeLocalDecl mvarId m!"\
+    local variable {Expr.fvar fvarId} is not present in local context{mvarId}"
+  let toRevert := lctx.foldl (init := #[]) fun arr decl' =>
+    if decl.index ≤ decl'.index then arr.push decl'.fvarId else arr
```

Comment:
@adomani If you want to be more comfortable reviewing, here's what to look for: the last five lines are new, creating this `toRevert` array.

The change to `runDefEqTactic` seems straighforward.

Reply:
Ok, I had indeed "micro-diffed" your PR, but thanks for the encouragement!

I'll be more assertive:

maintainer merge

Reply:
LGTM

bors merge